Patches.py

```diff
@@ -1562,16 +1562,21 @@ def update_scrub_text(message, text_replacement, default_price, price, item_name
         rom.write_byte(0xE209FD, 0x3C)
 
     if world.shuffle_medigoron_carpet_salesman:
-        rom.write_byte(rom.sym('SHUFFLE_CARPET_SALESMAN'), 0x01)
-        # Update carpet salesman messages to better fit the fact that he sells a randomized item
-        update_message_by_id(messages, 0x6077, "\x06\x41Well Come!\x04I am selling stuff, strange and \x01rare, from all over the world to \x01everybody.\x01Today's special is...\x04A mysterious item! \x01Intriguing! \x01I won't tell you what it is until \x01I see the money....\x04How about \x05\x41200 Rupees\x05\x40?\x01\x01\x1B\x05\x42Buy\x01Don't buy\x05\x40\x02")
-        update_message_by_id(messages, 0x6078, "Thank you very much!\x04The mark that will lead you to\x01the Spirit Temple is the \x05\x41flag on\x01the left \x05\x40outside the shop.\x01Be seeing you!\x02")
-
-        rom.write_byte(rom.sym('SHUFFLE_MEDIGORON'), 0x01)
-        # Update medigoron messages to better fit the fact that he sells a randomized item
-        update_message_by_id(messages, 0x304C, "I have something cool right here.\x01How about it...\x07\x30\x4F\x02")
-        update_message_by_id(messages, 0x304D, "How do you like it?\x02")
-        update_message_by_id(messages, 0x304F, "How about buying this cool item for \x01200 Rupees?\x01\x1B\x05\x42Buy\x01Don't buy\x05\x40\x02")
+        if not world.get_location('Wasteland Bombchu Salesman').item.name in ['Bombchus', 'Bombchus (5)', 'Bombchus (10)', 'Bombchus (20)']:
```

Comment:
Not sure I understand why we treat these as vanilla if they're shuffled and have something like their original item (Bombchus for Salesman, Giant's Knife for Medigoron).

Reply:
I thought it should be that way because the original text applies, but maybe they might want it to be a surprise